src/Mailgun/Connection/RestClient.php

```diff
@@ -25,30 +25,59 @@ class RestClient
     private $apiKey;
 
     /**
-     * @var Guzzle
+     * @var HttpAdapter
      */
-    protected $mgClient;
+    protected $httpAdapter;
+
+    /**
+     * @var string
+     */
+    protected $apiEndpoint;
 
     /**
      * @param string $apiKey
-     * @param string $apiEndpoint
+     * @param string $apiHost
      * @param string $apiVersion
      * @param bool   $ssl
      */
-    public function __construct($apiKey, $apiEndpoint, $apiVersion, $ssl)
+    public function __construct($apiKey, $apiHost, $apiVersion, $ssl, HttpAdapter $adapter = null)
     {
         $this->apiKey = $apiKey;
-        $this->mgClient = new Guzzle([
-            'base_url'=>$this->generateEndpoint($apiEndpoint, $apiVersion, $ssl),
-            'defaults'=>[
-                'auth' => array(Api::API_USER, $this->apiKey),
-                'exceptions' => false,
-                'config' => ['curl' => [ CURLOPT_FORBID_REUSE => true ]],
```

Comment:
Here is the commit that added the CURLOPT_FORBID_RESUSE.
https://github.com/mailgun/mailgun-php/commit/b4b9f0ec4018db76670e0beceff9e1021880d58c

Ping @travelton, I know it was over 2 years ago, but do you remember why you added that?